libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -324,18 +296,37 @@ + (BOOL)isNetworkError:(nullable NSError *)error {
 }
 
 #pragma mark SFAuthenticationManagerDelegate
-
-- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
+- (void)userAccountManagerDidChangeUserDataNotification:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+
     if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
-        if (![accountManager.currentUserIdentity isEqual:self.enqueuedNetwork.account.accountIdentity]) {
-            self.enqueuedNetwork.networkSuspended = YES;
-        } else {
-            [self.enqueuedNetwork resetSession];
-            self.enqueuedNetwork.networkSuspended = NO;
-        }
-        if (accountManager.currentCommunityId != self.enqueuedNetwork.defaultConnectCommunityId) {
-            self.enqueuedNetwork.defaultConnectCommunityId = accountManager.currentCommunityId;
+
+        NSString *userId = notification.userInfo[SFUserAccountManagerUserChangeUserIdKey];
+        NSString *orgId = notification.userInfo[SFUserAccountManagerUserChangeOrgIdKey];
+        SFUserAccountIdentity *identity = [[SFUserAccountIdentity alloc] initWithUserId:userId orgId:orgId];
+        SFUserAccount *userAccount = [[SFUserAccountManager sharedInstance] userAccountForUserIdentity:identity];
+        SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
+
+        if (change & SFUserAccountChangeCurrentUser) {
+            if (![accountManager.currentUserIdentity isEqual:self.enqueuedNetwork.account.accountIdentity]) {
+                self.enqueuedNetwork.networkSuspended = YES;
```

Comment:
We shouldn't ever set the `networkSuspended` property internally; this is meant to be used by the developer consuming the network stack to suspend/resume the network on their own.

This will also break instances where clean-up operations, or background operations need to be performed on one user while the account manager switches to a different user identity.